control/pid.go

```diff
@@ -63,9 +63,6 @@ func (p *basicPID) Next(ctx context.Context, x []*Signal, dt time.Duration) ([]*
 	} else {
 		dtS := dt.Seconds()
 		pvError := x[0].GetSignalValueAt(0)
-		if (p.sat > 0 && pvError > 0) || (p.sat < 0 && pvError < 0) {
-			return p.y, false
-		}
```

Comment:
I'm not at all confident this is a good solution. I tried removing it and it worked but I'm not sure why. I was reading [this wikipedia page about integral windup](https://en.wikipedia.org/wiki/Integral_windup) and looking at the "Solutions" section. I thought that bullet 4 was what the switch statement below this was doing, and I thought that these removed lines seemed similar to bullet 6, so I removed them because maybe we don't need two solutions to the same problem and the were interfering with each other? Then I read bullet 6 closer and I'm not actually confident that's what these lines were doing but removing them solved the problem so I thought a discussion here might be helpful.

Reply:
I agree, I do not think we need/want this if statement. Currently the way it(appears to me) behaves is once the controller saturates, the PID loop will be disabled until we overshoot the setpoint. Even if the integral term saturates, the control loop should still attempt to update the PD terms. 

I also think the if statement does not implement 6 either, and I don't necessarily think we should have 6 in cases where we use velocity control. 


I am also not sure what happens when we [do not update the channel](https://github.com/viamrobotics/rdk/blob/12e6cdcc994a998f175e31bb5f8a3f0597673713/control/control_loop.go#L150-L155) when that if statement triggered. talked about it abit with martha but maybe someone else has more context on what should happen?

as a side note, could maybe see changing the [integral saturation term](https://github.com/viamrobotics/rdk/blob/12e6cdcc994a998f175e31bb5f8a3f0597673713/control/setup_control.go#L266-L267) in our defaults to something lower, that way the proportional term should always have some affect on the motors. def not something we need to do tho, I personally just like being able to see the proportional term

Reply:
Your are correct to remove these lines the code is wrong :) . I think I initially meant it to signal the PID was saturating so action should be taken downstream to reduce the integral error with a proper anti-windup technique but I never got around doing it. The issue with saturating the integral term without correcting it is that it will take a while the the integral term to diminish if the setpoint is drastically changed I guess the question for you is when the base is saturating in one direction is changing the setpoint to an opposite value has an acceptable response time and behavior, if not we need to implement an anti-windup term happy to help with that.

If the channel is not updated it should use the last set value until it can be updated again.

Changing the integration saturation term will change the maximum correction we can attain. As long as| kP*e + kD*e + Isat  |> max then we are good but if we go below max (saturation of the actuator) then we are loosing in precision.

Reply:
In the scenario you mentioned where the base is saturating in one direction and then the direction is changed, the behavior seem fine to me it doesn't take an abnormal amount of time for the direction to switch. I wonder if that would take longer if the base was running for an extended period of time but based on this setup it seems like that shouldn't make a difference.

Reply:
then i think we don't need anti-windup. The integral is saturating so a longer time will not affect the response time (above a couple seconds)